### DISCUSSION TITLE
Soft cap on the number of derived addresses

### DISCUSSION THREAD

**Discussion by SlowestTimelord** [Chia-Network/chia-blockchain#14428] [Ideas]
Idea: add a parameter in the config to allow users to set the maximum number of derived addresses to use.
If set to 1, only 1 receive address will ever be shown and all unspent transaction change will be sent to the single sending address (similar to how Goby works).
If set to say 100, cycle back to the first derived address upon reaching derivation index 101.

At the cost of reduced privacy, this could solve performance issues with derivation index being too high (e.g. for NFT creator wallets).

**Comment by digitalspaceport** [comment]
Does this impact splitting of coins if it is set to 1 or is there another method to address that? I do like the idea if it can not impact functionality. Also concerned for impacts on performance of the local wallet potentially.

**Reply by braunswag** [reply]
This would make it a lot easier to track transactions with 1-5 wallet addresses

**Reply by SlowestTimelord** [reply]
I can't imagine it would impact coin split as it works fine in Goby.
